src/main/java/com/breskul/bibernate/persistence/EntityManagerImpl.java

```diff
@@ -1,29 +1,106 @@
 package com.breskul.bibernate.persistence;
 
+import com.breskul.bibernate.annotation.Id;
+import com.breskul.bibernate.annotation.JoinColumn;
+import com.breskul.bibernate.exception.JdbcDaoException;
 import com.breskul.bibernate.persistence.util.DaoUtils;
-import jakarta.persistence.*;
+import jakarta.persistence.EntityGraph;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.FlushModeType;
+import jakarta.persistence.LockModeType;
+import jakarta.persistence.Query;
+import jakarta.persistence.StoredProcedureQuery;
+import jakarta.persistence.TypedQuery;
 import jakarta.persistence.criteria.CriteriaBuilder;
 import jakarta.persistence.criteria.CriteriaDelete;
 import jakarta.persistence.criteria.CriteriaQuery;
 import jakarta.persistence.criteria.CriteriaUpdate;
 import jakarta.persistence.metamodel.Metamodel;
 
 import javax.sql.DataSource;
+import java.lang.reflect.Field;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
 import java.util.Map;
+import java.util.Queue;
 
-public class EntityMangerImpl implements EntityManager {
+import static com.breskul.bibernate.persistence.util.DaoUtils.getFieldByAnnotation;
+import static com.breskul.bibernate.persistence.util.DaoUtils.getFieldValue;
+import static com.breskul.bibernate.persistence.util.DaoUtils.isEntityCollectionField;
+import static com.breskul.bibernate.persistence.util.DaoUtils.isEntityField;
+import static com.breskul.bibernate.persistence.util.DaoUtils.isRegularField;
+import static com.breskul.bibernate.persistence.util.DaoUtils.isValidEntity;
+import static com.breskul.bibernate.persistence.util.DaoUtils.resolveFieldName;
+
+public class EntityManagerImpl implements EntityManager {
     private final JdbcDao jdbcDao;
 
-    public EntityMangerImpl(DataSource dataSource) {
+    public EntityManagerImpl(DataSource dataSource) {
         this.jdbcDao = new JdbcDao(dataSource);
     }
 
     @Override
-    public void persist(Object entity) {
+    public void persist(Object entityToSave) {
+        isValidEntity(entityToSave.getClass());
+        List<String> columns = new ArrayList<>();
+        List<Object> values = new ArrayList<>();
+        Queue<Collection<?>> queue = new ArrayDeque<>();
+        for (Field declaredField : entityToSave.getClass().getDeclaredFields()) {
+            if (declaredField.isAnnotationPresent(Id.class)) {
+                Field idField = getFieldByAnnotation(entityToSave.getClass(), Id.class);
+                Object entityId = jdbcDao.resolveEntityId(entityToSave, idField);
+                if (entityId != null) {
+                    processRegularField(columns, values, idField, entityId);
+                }
+                continue;
```

Comment:
I think better use else construction and avoid continue usage
![image](https://user-images.githubusercontent.com/32467464/228071379-394f0538-b3a8-4212-a9f1-c6063f3ef125.png)